sklearn/metrics/_classification.py

```diff
@@ -463,6 +464,23 @@ def confusion_matrix(
     >>> (tn, fp, fn, tp)
     (0, 2, 1, 1)
     """
+    xp, _, device_ = get_namespace_and_device(y_true, y_pred, labels, sample_weight)
+    y_true = check_array(
```

Comment:
Do you remember why you need this additional `check_array` here (and on the next statement), maybe this was related to old `pandas` version e.g. https://github.com/scikit-learn/scikit-learn/pull/30562#issuecomment-2586855176?